metals/src/main/scala/scala/meta/internal/metals/watcher/SimpleFileWatcher.scala

```diff
@@ -0,0 +1,66 @@
+package scala.meta.internal.metals.watcher
+
+import collection.JavaConverters._
+import java.nio.file.{FileSystems, Path, StandardWatchEventKinds}
+import java.util.concurrent.BlockingQueue
+import scala.concurrent.{ExecutionContext, Future}
+
+object SimpleFileWatcher {
+  /*
+   * returns a function which closes the file watcher when called
+   */
+  def initWatcher(
+    watchFilter: Path => Boolean,
+    watchEventQueue: BlockingQueue[FileWatcherEvent],
+    filesToWatch: Set[Path],
+    directoriesToWatch: Set[Path],
+  )(implicit ec: ExecutionContext): () => Unit = {
+    scribe.error(s"RFERGUSON INIT WATCHER")
+    val watchService = FileSystems.getDefault().newWatchService();
+    filesToWatch.filter(watchFilter).foreach { path =>
+      scribe.info(s"watching parent directory of file $path")
+      // watchservice only watches directories
+      // this technically could watch things that weren't in the build, but shared a directory, but what can you do?
```

Comment:
We could listen to only modifications and filter those that correspond to files in `filesToWatch` before we send them to `watchEventQueue`.